src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,487 @@
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.isUnlocked = value
+        }
+
+    var slot: Slot? = null
+        private set
+
+    fun getLevelUpCost() = costFun(rawLevel)
+
+    fun getReward() = rewardFun(activeLevel)
+
+    fun calculateTotalCost(desiredLevel: Int) = (0 until desiredLevel).sumOf { level -> costFun(level) ?: 0.0 }.toInt()
+
+    companion object {
+
+        val storage get() = ProfileStorageData.profileSpecific?.mining?.hotmTree
+
+        val abilities =
+            listOf(PICKOBULUS, MINING_SPEED_BOOST, VEIN_SEEKER, MANIAC_MINER, HAZARDOUS_MINER, GEMSTONE_INFUSION)
+
+        private val inventoryPattern by repoGroup.pattern(
+            "inventory", "Heart of the Mountain"
+        )
+
+        private val levelPattern by repoGroup.pattern(
+            "perk.level", "§(?<color>.)Level (?<level>\\d+).*"
+        )
+
+        private val notUnlockedPattern by repoGroup.pattern(
+            "perk.notunlocked", "(§.)*Requires.*|.*Mountain!|(§.)*Click to unlock!|"
+        )
+
+        private val enabledPattern by repoGroup.pattern(
+            "perk.enable", "§a§lENABLED|(§.)*SELECTED"
+        )
+        private val disabledPattern by repoGroup.pattern(
+            "perk.disabled", "§c§lDISABLED|§7§eClick to select!"
+        ) // unused for now since the assumption is when enabled isn't found it is disabled, but the value might be useful in the future or for debugging
+
+        private val resetChatPattern by repoGroup.pattern(
+            "reset.chat", "§aReset your §r§5Heart of the Mountain§r§a! Your Perks and Abilities have been reset."
+        )
+
+        private val heartItemPattern by repoGroup.pattern(
+            "inventory.heart", "§5Heart of the Mountain"
+        )
+        private val resetItemPattern by repoGroup.pattern(
+            "inventory.reset", "§cReset Heart of the Mountain"
+        )
+
+        private val heartTokensPattern by repoGroup.pattern(
+            "inventory.heart.token", "§7Token of the Mountain: §5(?<token>\\d+)"
+        )
+
+        private val resetTokensPattern by repoGroup.pattern(
+            "inventory.reset.token", "\\s+§8- §5(?<token>\\d+) Token of the Mountain"
+        )
+
+        var inInventory = false
+
+        var tokens: Int
+            get() = ProfileStorageData.profileSpecific?.mining?.tokens ?: 0
+            private set(value) {
+                ProfileStorageData.profileSpecific?.mining?.tokens = value
+            }
+
+        var availableTokens: Int
+            get() = ProfileStorageData.profileSpecific?.mining?.availableTokens ?: 0
+            private set(value) {
+                ProfileStorageData.profileSpecific?.mining?.availableTokens = value
+            }
+
+        var heartItem: Slot? = null
+
+        init {
+            entries.forEach {
+                it.guiNamePattern
+                it.totalCostMaxLevel = it.calculateTotalCost(it.maxLevel)
```

Comment:
this assignment can be done in the enum. No need to have it in the companion object. Also makes it possible to make total Cost Maxlevel a val.

Reply:
This would be worse because instead of being able to access the calculateTotalCost() method, I would need to write it out every single time, and I wouldn't even be able to use the maxLevel constant in the enum entry itself.
I might be overlooking some things right now, please tell me if I do so.

Reply:
I don't see a problem here. Since calculateTotalCost is known at the instantiation. So why shouldn't it work? Maybe try change the order of the elments eg. calculateTotalCost and after that total cost. And inside the enum you can use the maxlevel.
So there is no issue here. Maybe I worded it missleading.
So here literally the code:
```Kotlin
val maxlevelCost = calculateTotalCost(maxLevel)
```

Reply:
When I try to set the max level inside of the enum constructor, the method won't get detected.
Maybe I still don't understand what you mean, please explain it to me then.
![image](https://github.com/hannibal002/SkyHanni/assets/45315647/c16a642d-86f3-4520-bc83-21374dbe64ab)

Reply:
that is not in the enum, that is inside the constructor of the enum. Inside the enum is after ;

Reply:
![image](https://github.com/hannibal002/SkyHanni/assets/85900443/2bdda70d-477f-406a-add7-90e57d55234b)

Reply:
* as a val please. Forgot to change that.

Reply:
Oh my fucking god.
Sorry I'm stupid
Thanks for explaining!